drivers/gpio/gpio_nrfx.c

```diff
@@ -492,16 +492,13 @@ static void gpiote_event_handler(void)
 						 NRF_GPIOTE_EVENT_PORT);
 
 	if (port_event) {
-#ifdef CONFIG_GPIO_NRF_P0
-		fired_triggers[0] =
-			check_level_trigger_pins(DEVICE_DT_GET(GPIO(0)),
-						 &sense_levels[0]);
-#endif
-#ifdef CONFIG_GPIO_NRF_P1
-		fired_triggers[1] =
-			check_level_trigger_pins(DEVICE_DT_GET(GPIO(1)),
-						 &sense_levels[1]);
-#endif
+		#define GPIO_NRF_GET_TRIGGERS(id) \
+			fired_triggers[id] = \
```

Comment:
Since `gpio0` could be potentially disabled, I think it needs to be:
```suggestion
			fired_triggers[DT_INST_PROP(id, port)] = \
```
because of: https://github.com/zephyrproject-rtos/zephyr/blob/8539a53e3d860f9d241f22e071b79f551c7600ca/drivers/gpio/gpio_nrfx.c#L516-L518

Reply:
done, to confirm tried running gpio_basic_api test on nrf52840dk with `gpio0` disabled since this test is using pins from port 1.